tests/glsl/ssboMemberMemoryQualifier.slang

```diff
@@ -0,0 +1,48 @@
+//DIAGNOSTIC_TEST:SIMPLE(filecheck=CHECK-GLSL): -stage compute -entry computeMain -allow-glsl -target glsl
+//DIAGNOSTIC_TEST:SIMPLE(filecheck=CHECK-SPV): -stage compute -entry computeMain -allow-glsl -target spirv -emit-spirv-directly
```

Comment:
DIAGNOSTIC_TEST is just syntax candy. I was using it since I was basing this file off of the error check files I made which were based on a file that use DIAGNOSTIC_TEST ('unknown-opcode.slang` is an example of this).

We can have a runtime test; we don't really need it though since the goal is "make sure we can add qualifiers here" as an opposite to "make sure we cannot add qualifier here" tests. Testing if we can write 1 to a int is not really helpful since the regression is all "compiler enforcing things" and decorations which indirectly effect runtime. If glslang does not compile glslang would complain, if glslang compiles, glslang won't complain.

Reply:
I still don't know what DIAGNSTIC_TEST is.
Can you tell me why this test uses DIAGNSTIC_TEST not TEST?

If the test is not meant for runtime result, we don't need to test it here. That is fine.

FYI, the other day, Yong told me that we cannot rely on the error messages on the downstream compilers. Slang must catch and report the syntactic errors without relying on the error messages from glslang.

Reply:
1. "Can you tell me why this test uses DIAGNSTIC_TEST not TEST?" because the code I based this code off of used DIAGNSTIC_TEST, they function identical, I will switch to TEST, there is no issue in doing so 

2.👍 

3. we don't rely on glslang error messages for any of the following code, I mean it as in: this is all compile time syntax candy for as far as the programmer will experience, the compiler (Slang) will error if you break syntax rules (and I am checking for these)

Reply:
It looks like DIAGNOSTIC_TEST is meant for testing error cases.
```
struct TestOptions
{
    enum Type
    {
        Normal,             ///< A regular test
        Diagnostic,         ///< Diagnostic tests will always run (as form of failure is being tested)
    };
```
So if your intention is to testing failures or error messages, it seems appropriate to use it.